Add dev shell environment (#1534)
stellar/stellar-cli: CONTRIBUTING.md

  * Clearly describe the issue including steps to reproduce if it is a bug.
* Fork the repository on GitHub.

## Setting up development environment

There are 2 ways to being developing stellar-cli:

### Installing all required dependencies 

> You may want to install all required dependencies locally. This includes installing `rustup`, `make`, `libudev`, `jq` from your package manager. After all dependencies are installed, you can start with running `make build` to build `stellar-cli` on your machine! 

Reviewer: ```suggestion
You may want to install all required dependencies locally. This includes installing `rustup`, `make`, `libudev`, `jq` from your package manager. After all dependencies are installed, you can start with running `make install` to build `stellar-cli` and install it! 
```
